spec.html

```diff
@@ -149,6 +149,39 @@ <h1>
         1. Return _iteratorRecord_.
       </emu-alg>
     </emu-clause>
+
+    <emu-clause id="sec-getiteratorflattenable" type="abstract operation">
+      <h1>
+        GetIteratorFlattenable (
+          _obj_: an ECMAScript language value,
+          _hint_: ~sync~ or ~async~,
+        ): either a normal completion containing an Iterator Record or a throw completion
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. If _obj_ is not an Object, throw a *TypeError* exception.
+        1. Let _alreadyAsync_ be *false*.
+        1. Let _method_ be *undefined*.
+        1. If _hint_ is ~async~, then
+          1. Set _method_ to ? Get(_obj_, @@asyncIterator).
+          1. Set _alreadyAsync_ to *true*.
+        1. If IsCallable(_method_) is *false*, then
+          1. Set _method_ to ? Get(_obj_, @@iterator).
+          1. Set _alreadyAsync_ to *false*.
+        1. If IsCallable(_method_) is *false*, then
+          1. Let _iterator_ be _obj_.
+          1. Set _alreadyAsync_ to *true*.
```

Comment:
i think this variable name is confusing, since `_iterator_` isn't already async here, it just shouldn't be made async later (if i'm understanding right)

Reply:
This is only applicable in the `_hint_ is ~async~` path. The iterator is assumed to be async if we're trying to get an async iterator. I could have guarded this assignment for clarity, but it wouldn't change the result.